molgenis-i18n/src/main/java/org/molgenis/i18n/CodedRuntimeException.java

```diff
@@ -0,0 +1,53 @@
+package org.molgenis.i18n;
+
+import java.text.MessageFormat;
+
+import static java.util.Objects.requireNonNull;
+import static org.molgenis.i18n.MessageSourceHolder.getMessageSource;
+import static org.springframework.context.i18n.LocaleContextHolder.getLocale;
+
+/**
+ * {@link RuntimeException} with error code.
+ */
+@SuppressWarnings("squid:MaximumInheritanceDepth")
+public abstract class CodedRuntimeException extends RuntimeException implements ErrorCoded
+{
+	private final String errorCode;
+
+	protected CodedRuntimeException(String errorCode)
+	{
+		this.errorCode = requireNonNull(errorCode);
+	}
+
+	protected CodedRuntimeException(String errorCode, Throwable cause)
+	{
+		super(cause);
+		this.errorCode = errorCode;
+	}
+
+	@Override
+	public String getErrorCode()
+	{
+		return errorCode;
+	}
+
+	@Override
+	public String getLocalizedMessage()
```

Comment:
Create a version of coded runtime exception that is independent of the languageservice for technical exceptions.
Maybe a LocalizedRuntimeException the extends from CodedRuntimeException

Reply:
assign

Reply:
proposal: keep as is until @bartcharbon and others are back from holidays